ios/Classes/dropIn/DropInPlatformApi.swift

```diff
@@ -221,8 +256,32 @@ class DropInPlatformApi: DropInPlatformInterface {
             hostViewController?.adyen.topPresenter.present(alertController, animated: true)
         }
     }
+    
+    private func onDropInResultUpdate(paymentEventDTO: PaymentEventDTO) {
+        do {
+            guard let updatedPaymentMethods = paymentEventDTO.data?[Constants.updatedPaymentMethodsKey] ?? "" else {
+                throw PlatformError(errorDescription: "Updated payment methods not provided.")
+            }
+            
+            guard let orderResponse = paymentEventDTO.data?[Constants.orderKey] ?? "" else {
+                throw PlatformError(errorDescription: "Order not provided.")
+            }
+            
+            let updatedPaymentMethodsData = try JSONSerialization.data(withJSONObject: updatedPaymentMethods, options: [])
+            let paymentMethods = try jsonDecoder.decode(PaymentMethods.self, from: updatedPaymentMethodsData)
+            let orderData = try JSONSerialization.data(withJSONObject: orderResponse, options: [])
+            let order = try jsonDecoder.decode(PartialPaymentOrder.self, from: orderData)
+            try dropInViewController?.dropInComponent.reload(with: order, paymentMethods)
+        } catch {
+            adyenPrint(error.localizedDescription)
+        }
+    }
 
-    private func removeGiftCardPaymentMethods(paymentMethods: PaymentMethods) -> PaymentMethods {
+    private func removeGiftCardPaymentMethods(paymentMethods: PaymentMethods, isPartialPaymentSupported: Bool) -> PaymentMethods {
+        guard !isPartialPaymentSupported else {
+            return paymentMethods
+        }
```

Comment:
I think this one is hard to read now 😅

Simple IF is better, since **guard** is for validation or edge-casese and here it is a  pure Business logic.
```swift
if isPartialPaymentSupported {
    return paymentMethods
}
```

Reply:
🙈